[MAINT]: `npm run build` is failing

### Describe the need

see https://github.com/octokit/auth-callback.js/actions/runs/9470231822/job/26090632310

@wolfy1339 I think you did the update to the build scripts across @octokit. Does that one ring any bells?

### SDK Version

_No response_

### API Version

_No response_

### Relevant log output

```
Run npm run build

> @octokit/auth-callback@0.0.0-development build
> node scripts/build.mjs && tsc -p tsconfig.json

Error: src/auth.ts(3,1): error TS1287: A top-level 'export' modifier cannot be used on value declarations in a CommonJS module when 'verbatimModuleSyntax' is enabled.
Error: src/hook.ts(11,10): error TS1286: ESM syntax is not allowed in a CommonJS module when 'verbatimModuleSyntax' is enabled.
Error: src/hook.ts(13,1): error TS1287: A top-level 'export' modifier cannot be used on value declarations in a CommonJS module when 'verbatimModuleSyntax' is enabled.
Error: src/index.ts(1,10): error TS1286: ESM syntax is not allowed in a CommonJS module when 'verbatimModuleSyntax' is enabled.
Error: src/index.ts(2,10): error TS1286: ESM syntax is not allowed in a CommonJS module when 'verbatimModuleSyntax' is enabled.
Error: src/index.ts(15,1): error TS1287: A top-level 'export' modifier cannot be used on value declarations in a CommonJS module when 'verbatimModuleSyntax' is enabled.
Error: src/types.ts(1,13): error TS1286: ESM syntax is not allowed in a CommonJS module when 'verbatimModuleSyntax' is enabled.
Error: src/with-authorization-prefix.ts(6,1): error TS1287: A top-level 'export' modifier cannot be used on value declarations in a CommonJS module when 'verbatimModuleSyntax' is enabled.
Error: Process completed with exit code 2.
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/package.json
+++ b/package.json
@@ -20,8 +20,8 @@
   "author": "Gregor Martynus (https://dev.to/gr2m)",
   "license": "MIT",
   "devDependencies": {
-    "@octokit/core": "^5.0.0",
-    "@octokit/request": "^8.0.0",
+    "@octokit/core": "^6.0.0",
+    "@octokit/request": "^9.0.0",
     "@octokit/tsconfig": "^3.0.0",
     "@types/jest": "^29.0.0",
     "@types/node": "^20.0.0",

--- a/package.json
+++ b/package.json
@@ -1,13 +1,14 @@
 {
   "name": "@octokit/auth-callback",
   "version": "0.0.0-development",
+  "type": "module",
   "description": "GitHub API authentication using a callback method",
   "scripts": {
     "build": "node scripts/build.mjs && tsc -p tsconfig.json",
     "lint": "prettier --check '{src,test,scripts}/**/*' README.md package.json",
     "lint:fix": "prettier --write '{src,test,scripts}/**/*' README.md package.json",
     "pretest": "npm run -s lint",
-    "test": "jest --coverage"
+    "test": "NODE_OPTIONS=\"$NODE_OPTIONS --experimental-vm-modules\" npx jest --coverage"
   },
   "repository": "github:octokit/auth-callback.js",
   "keywords": [
@@ -33,11 +34,15 @@
     "typescript": "^5.0.0"
   },
   "jest": {
+    "extensionsToTreatAsEsm": [
+      ".ts"
+    ],
     "transform": {
       "^.+\\.(ts|tsx)$": [
         "ts-jest",
         {
-          "tsconfig": "test/tsconfig.test.json"
+          "tsconfig": "test/tsconfig.test.json",
+          "useESM": true
         }
       ]
     },
@@ -48,6 +53,9 @@
         "functions": 100,
         "lines": 100
       }
+    },
+    "moduleNameMapper": {
+      "^(.+)\\.jsx?$": "$1"
     }
   },
   "release": {

--- a/scripts/build.mjs
+++ b/scripts/build.mjs
@@ -8,6 +8,9 @@ const sharedOptions = {
   minify: false,
   allowOverwrite: true,
   packages: "external",
+  platform: "neutral",
+  format: "esm",
+  target: "es2022",
 };
 
 async function main() {
@@ -18,8 +21,6 @@ async function main() {
     entryPoints: await glob(["./src/*.ts", "./src/**/*.ts"]),
     outdir: "pkg/dist-src",
     bundle: false,
-    platform: "neutral",
-    format: "esm",
     ...sharedOptions,
     sourcemap: false,
   });
@@ -35,27 +36,12 @@ async function main() {
 
   const entryPoints = ["./pkg/dist-src/index.js"];
 
-  await Promise.all([
-    // Build the a CJS Node.js bundle
-    esbuild.build({
-      entryPoints,
-      outdir: "pkg/dist-node",
-      bundle: true,
-      platform: "node",
-      target: "node14",
-      format: "cjs",
-      ...sharedOptions,
-    }),
-    // Build an ESM browser bundle
-    esbuild.build({
-      entryPoints,
-      outdir: "pkg/dist-web",
-      bundle: true,
-      platform: "browser",
-      format: "esm",
-      ...sharedOptions,
-    }),
-  ]);
+  await esbuild.build({
+    entryPoints,
+    outdir: "pkg/dist-bundle",
+    bundle: true,
+    ...sharedOptions,
+  });
 
   // Copy the README, LICENSE to the pkg folder
   await copyFile("LICENSE", "pkg/LICENSE");
@@ -74,10 +60,15 @@ async function main() {
       {
         ...pkg,
         files: ["dist-*/**", "bin/**"],
-        main: "dist-node/index.js",
-        browser: "dist-web/index.js",
         types: "dist-types/index.d.ts",
-        module: "dist-src/index.js",
+        exports: {
+          ".": {
+            types: "./dist-types/index.d.ts",
+            import: "./dist-bundle/index.js",
+            // Tooling currently are having issues with the "exports" field when there is no "default", ex: TypeScript, eslint, ncc
+            default: "./dist-bundle/index.js",
+          },
+        },
         sideEffects: false,
       },
       null,

--- a/src/auth.ts
+++ b/src/auth.ts
@@ -1,4 +1,4 @@
-import type { Callback, Authentication } from "./types";
+import type { Callback, Authentication } from "./types.js";
 
 export async function auth(callback: Callback): Promise<Authentication> {
   const result = await callback();

--- a/src/hook.ts
+++ b/src/hook.ts
@@ -6,9 +6,9 @@ import type {
   RequestParameters,
   Route,
   Callback,
-} from "./types";
+} from "./types.js";
 
-import { withAuthorizationPrefix } from "./with-authorization-prefix";
+import { withAuthorizationPrefix } from "./with-authorization-prefix.js";
 
 export async function hook(
   callback: Callback,

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,10 +1,10 @@
-import { auth } from "./auth";
-import { hook } from "./hook";
+import { auth } from "./auth.js";
+import { hook } from "./hook.js";
 import type {
   StrategyInterface,
   StrategyOption,
   Authentication,
-} from "./types";
+} from "./types.js";
 
 export type Types = {
   StrategyOptions: StrategyOption;

--- a/test/index.test.ts
+++ b/test/index.test.ts
@@ -2,7 +2,7 @@ import { request } from "@octokit/request";
 import fetchMock, { MockMatcherFunction } from "fetch-mock";
 import { Octokit } from "@octokit/core";
 
-import { createCallbackAuth } from "../src/index";
+import { createCallbackAuth } from "../src/index.js";
 
 test("README example", async () => {
   let token: string | undefined;

